File: Leetcode/13.Roman to Integer/Python/main.py
roman = input()
integer = 0
sz = len(roman)
for i in range(sz):
    if(roman[i]=="I"):
        integer += 1
    elif(roman[i]=="V"):
        integer += 5
    elif(roman[i]=="X"):
        integer += 10
    elif(roman[i]=="L"):
        integer += 50
    elif(roman[i]=="C"):
        integer += 100
    elif(roman[i]=="D"):
        integer += 500
    elif(roman[i]=="M"):
        integer += 1000
    if(not i==sz-1):
        if((roman[i]=='I' and roman[i+1]=='V') or (roman[i]=='I' and roman[i+1]=='X')):
                integer-=2
        if((roman[i]=='X' and roman[i+1]=='L') or (roman[i]=='X' and roman[i+1]=='C')):
                integer-=20
        if((roman[i]=='C' and roman[i+1]=='D') or (roman[i]=='C' and roman[i+1]=='M')):
                integer-=200
print(integer) 

File: Leetcode/13.Roman to Integer/C++/main.cpp
#include <bits/stdc++.h>
using namespace std;
map<char,int>roman_conversion;
void generate_map(){
    roman_conversion['I']=1;
    roman_conversion['V']=5;
    roman_conversion['X']=10;
    roman_conversion['L']=50;
    roman_conversion['C']=100;
    roman_conversion['D']=500;
    roman_conversion['M']=1000;
}
int main(){
    generate_map();
    string roman;
    int integer =0;
    cin>>roman;
    int sz = roman.length();
    for(int i=0;i<sz;){
        if(i == sz-1 || roman_conversion[roman[i]] >= roman_conversion[roman[i+1]]){
            integer+=roman_conversion[roman[i]];
            i++;
        }
        else{
            integer+= roman_conversion[roman[i+1]] - roman_conversion[roman[i]];
            i+=2;
        }
    }

    cout<<integer<<endl;
    return 0;
}